src/components/forms/NumberField/NumberField.stories.mdx

```diff
@@ -35,22 +43,118 @@ const generalTest = async canvasElement => {
   }}
 />
 
+# Number inputs
+
+Numeric inputs are essential for some types of data, but using the correct inputs is rather hard due
+to a couple of reasons:
+
+- some data looks numeric, but actually isn't
+- associated validations (minimum and maximum values)
+- accessibility concerns (especially when silently truncating or ignoring data)
+- localisation - different locales use different decimal separators
+
+The implementation of our `Number` component attempts to address these, and has taken certain
+sources into account when doing so:
+
+- https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/
+
+## SSN, Credit card, phone numbers...
+
+Do not use the `NumberField` for these, instead use a `TextField`, possibly with
+`inputmode="numeric"` and `pattern=[0-9]+` props.
+
+This kind of data are actually just text inputs/strings that happen to consist only of digits. As a
+rule of thumb, if leading zeroes are significant, you should use a `TextField` instead.
+
+## Small number inputs
+
+When you expect small, incremental values for numbers (such as number of attendees or products), you
+can use this variant by specyfing the `useNumberType` boolean prop. This results in increment/
+decrement buttons being exposed to the user so they don't need to type in the value.
+
+This works well together with the `min` prop to specify a minimum value and `step` to specify the
+increment/decrement step. E.g. to cycle between positive integers, you would use `min={1} step={1}`.
+
+**Warning**
+
+The HTML input does not prevent people from typing in invalid values (such as negative numbers or
+decimals). You need to address this with input validators at the form level and display the
+appropriate validation error(s).
+
+(Silently) truncating user input (like a minus sign or decimal separator) is confusing, and for
+users of assistive technology even worse as they have no feedback on what is actually
+accepted/displayed on the screen.
+
+<Canvas>
+  <Story
+    name="Small numbers with incr/decr buttons"
+    args={{
+      name: 'amount',
+      id: 'amount-type-number',
+      label: 'Amount',
+      description: 'HTML type=number, still allows typing invalid values.',
+      useNumberType: true,
+      disabled: false,
+      invalid: false,
+      isRequired: false,
+    }}
+    play={async ({canvasElement}) => {
+      const canvas = within(canvasElement);
+      const input = canvas.getByLabelText('Amount');
+      // assert that we can still type 'large' numbers
+      await userEvent.clear(input);
+      await userEvent.type(input, '1000');
+      await expect(input).toHaveDisplayValue('1000');
+      // assert that we can type floats without localisation
+      await userEvent.clear(input);
+      await userEvent.type(input, '1.5');
+      await expect(input).toHaveDisplayValue('1.5');
+    }}
```

Comment:
Note that I didn't apply the `generalTest` thing here.

I am personally of the philosophy that tests should be 100% self-contained and calling out to subtests/functions etc. makes it harder to debug/check tests on what is going on.

Additionally, I typically rely on the `canvas.getByLabelText` to ensure that label + input are correctly linked, which then also implicitly proves that clicking the label focuses the input. So, for me, those `expect(...).toBeFocused()` tests do not need to written out, but it doesn't hurt to do so either.